src/test/scala/uk/gov/ons/bi/dataload/ubrn/LinksPreprocessorFlatSpec.scala

```diff
@@ -0,0 +1,63 @@
+package uk.gov.ons.bi.dataload.ubrn
+
+import org.scalatest.{FlatSpec, Matchers}
+import uk.gov.ons.bi.dataload.model.BiSparkDataFrames
+import uk.gov.ons.bi.dataload.SparkCreator
+
+class LinksPreprocessorFlatSpec extends FlatSpec with Matchers with SparkCreator {
+
+  import spark.implicits._
+
+  val lpp = new LinksPreprocessor(ctxMgr)
+
+  "Links Preprocessor" should "Read set previous Links to empty Dataframe if error" in {
+    val emptyDF = lpp.readPrevLinks("invalidDir","invalidFile")
+
+    emptyDF.take(1).isEmpty shouldBe true
+  }
+
+  "Links Preprocessor" should "apply UBRN to Dataframe input" in {
+
+    val data = Seq(
+      (Array("ch1"), Array(""), Array("065H7Z31732"))
+    ).toDF("CH","VAT","PAYE")
+
+    val prevData = BiSparkDataFrames.emptyLinkWithUbrnDf(ctxMgr)
+
+    val actual = lpp.preProcessLinks(data, prevData)
+
+    val expected = Seq(
+      (1000000000000001L, Array("ch1"), Array(""), Array("065H7Z31732"))
+    ).toDF("UBRN","CH","VAT","PAYE")
+
+    actual.collect() shouldBe expected.collect()
```

Comment:
It would be nice to have a custom assertion here that calls the collect for you. Something like `assertSmallDataFrameEquality(actualDF, expectedDF)`